tools/legacy/sample_encode/src/pipeline_encode.cpp

```diff
@@ -2387,6 +2393,12 @@ mfxStatus CEncodingPipeline::EncodeOneFrame(const ExtendedSurface &In, sTask *&p
             MSDK_IGNORE_MFX_STS(sts, MFX_ERR_MORE_BITSTREAM);
             if (m_nPerfOpt)
                 m_nEncSurfIdx++;
+
+            if (m_bAPI2XInternalMem)
```

Comment:
It seems checking "In.pSurface!=nullptr" is needed here (in case of flashing input surface maybe empty).